### PR TITLE
GUI improvement

### PR DESCRIPTION
**Pull request by egecansen**
- Improved modularity

### CODE REVIEW

File: src/main/java/gpt/chat/ui/theme/SupportGUIDark.java [hunk 1]
```diff
@@ -120,38 +120,37 @@ public SupportGUIDark(
...
         chatOverviewScrollPane.getVerticalScrollBar().setUI(new BasicScrollBarUI() {
-            private static final Color THUMB_COLOR = new Color(46 , 46 ,52);
-            private static final Color TRACK_COLOR = new Color(67 , 69 ,74);
 
             @Override
             protected void paintThumb(Graphics g, JComponent c, Rectangle thumbBounds) {
                 Graphics2D g2 = (Graphics2D) g.create();
                 g2.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
-                g2.setColor(THUMB_COLOR);
+                g2.setColor(thumbColors);
```
Comment (Umutayb) [inline src/main/java/gpt/chat/ui/theme/SupportGUIDark.java:153]: Remove plural (s)

File: src/main/java/gpt/chat/ui/theme/SupportGUIDark.java [hunk 2]
```diff
@@ -160,7 +159,7 @@ protected void paintThumb(Graphics g, JComponent c, Rectangle thumbBounds) {
             protected void paintTrack(Graphics g, JComponent c, Rectangle trackBounds) {
                 Graphics2D g2 = (Graphics2D) g.create();
                 g2.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
-                g2.setColor(TRACK_COLOR);
+                g2.setColor(trackColors);
```
Comment (Umutayb) [inline src/main/java/gpt/chat/ui/theme/SupportGUIDark.java:162]: Remove plural (s)

File: src/main/java/gpt/chat/ui/theme/SupportGUIDark.java [hunk 3]
```diff
@@ -185,14 +184,12 @@ private JButton createZeroButton() {
...
         messageInputScrollPane.getVerticalScrollBar().setUI(new BasicScrollBarUI() {
-            private static final Color THUMB_COLOR = new Color(46 , 46 ,52);
-            private static final Color TRACK_COLOR = new Color(67 , 69 ,74);
 
             @Override
             protected void paintThumb(Graphics g, JComponent c, Rectangle thumbBounds) {
                 Graphics2D g2 = (Graphics2D) g.create();
                 g2.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
-                g2.setColor(THUMB_COLOR);
+                g2.setColor(thumbColors);
```
Comment (Umutayb) [inline src/main/java/gpt/chat/ui/theme/SupportGUIDark.java:192]: Remove plural (s)

File: src/main/java/gpt/chat/ui/theme/SupportGUIDark.java [hunk 4]
```diff
@@ -201,7 +198,7 @@ protected void paintThumb(Graphics g, JComponent c, Rectangle thumbBounds) {
             protected void paintTrack(Graphics g, JComponent c, Rectangle trackBounds) {
                 Graphics2D g2 = (Graphics2D) g.create();
                 g2.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
-                g2.setColor(TRACK_COLOR);
+                g2.setColor(trackColors);
```
Comment (Umutayb) [inline src/main/java/gpt/chat/ui/theme/SupportGUIDark.java:201]: Remove plural (s)

File: src/main/java/gpt/chat/ui/theme/SupportGUIDark.java [hunk 5]
```diff
@@ -226,89 +223,79 @@ private JButton createZeroButton() {
...
+        }
+    }
 
-                    if (e.getKeyCode() == KeyEvent.VK_DOWN) {
-                        String currentMessage = messageInputPanel.getText().trim();
-                        messageInputPanel.setText(oldMsg);
-                        oldMsg = currentMessage;
-                    }
+    @SuppressWarnings("UnusedReturnValue")
+    public JFrame setUpSupportPanel() {
```
Comment (Umutayb) [inline src/main/java/gpt/chat/ui/theme/SupportGUIDark.java:252]: Should be void

File: src/main/java/gpt/chat/ui/theme/SupportGUILight.java [hunk 6]
```diff
@@ -124,80 +125,119 @@ public SupportGUILight(
...
+    }
 
-                    // Get last message typed
-                    if (e.getKeyCode() == KeyEvent.VK_UP) {
-                        String currentMessage = messageInputPanel.getText().trim();
-                        messageInputPanel.setText(oldMsg);
-                        oldMsg = currentMessage;
-                    }
+    @SuppressWarnings("UnusedReturnValue")
+    public JFrame setUpSupportPanel() {
```
Comment (Umutayb) [inline src/main/java/gpt/chat/ui/theme/SupportGUILight.java:152]: Should be void!

**Review by Umutayb — CHANGES_REQUESTED** [review]
Requested changes!

File: src/main/java/gpt/chat/ui/theme/SupportGUIDark.java (hunk 1 above)
Comment (egecansen) [reply src/main/java/gpt/chat/ui/theme/SupportGUIDark.java]: change it to thumbColour/trackColour

File: src/main/java/gpt/chat/ui/theme/SupportGUIDark.java (hunk 5 above)
Comment (egecansen) [reply src/main/java/gpt/chat/ui/theme/SupportGUIDark.java]: Done!

File: src/main/java/gpt/chat/ui/theme/SupportGUIDark.java (hunk 2 above)
Comment (egecansen) [reply src/main/java/gpt/chat/ui/theme/SupportGUIDark.java]: Done!

**Review by Umutayb — APPROVED** [review]

**Pull request merged**